crypto/sig/README.md

```diff
@@ -77,10 +75,13 @@ The [ed25519 manuscript]((http://ed25519.cr.yp.to/ed25519-20110926.pdf)) from 20
 > a cipher such as AES, combined with standard PRF-stretching mechanisms to support a long input;
 > but we prefer to reuse `H` to save area in hardware implementations.
 
-We point out that Blake was one of the SHA-3 finalists, and was also explicitly designed to model a PRF.
-We prefer Blake2b here because it reduces the number of different hash functions in our system overall.
+We point out that Merlin Transcripts implement a subset of the STROBE protocol, which states in [STROBE Protocol Framework](https://strobe.sourceforge.io/),
 
-Assuming the Blake2b has the secret-prefix-PRF property, we can say that the signatures created this way
+>  Strobe is based on SHA-3, or rather Keccak-f and cSHAKE (draft NIST SP 800-185).
+
+We prefer using Merlin Transcripts here for simplicity, as we are constructing two transcripts in the course of constructing the signature, and because it reduces the number of different hash functions in our system overall.
+
+Assuming the Merlin `challenge_bytes` has the secret-prefix-PRF property, we can say that the signatures created this way
```

Comment:
When using Merlin for this, instead of assuming that a hash function has the secret-prefix-PRF property, we assume that the following pseudo-rust function does, for any particular merlin transcript, where the private_key argument is identified with the secret prefix:

```
fn produce_nonce(transcript, private_key, message) -> [u8; 32]
    transcript.append_message("private", private_key);
    transcript.append_message("message", message);
    let mut nonce = [0u8; 32];
    transcript.challenge_bytes("nonce", &mut nonce);
    nonce
}
```

This assumption can be justified if we believe that the STROBE "PRF" operation functions as a PRF once STROBE has been keyed. We refer the reader to https://strobe.sourceforge.io/papers/strobe-20170130.pdf for a discussion of the usage and security properties of STROBE. For more discussion, see also the documentation around Merlin: https://merlin.cool/transcript/ops.html.

With this assumption in hand, we can say that signatures created this way are hard to distinguish from signatures ...

Reply:
Excellent - ty!